### PR TITLE
Check genesis block during recovery.

### PR DESCRIPTION
**Pull request by vldm**
Closes: #483

### CODE REVIEW

File: node/src/lib.rs [hunk 1]
```diff
@@ -361,6 +361,13 @@ impl NodeService {
         if self.chain.blocks().len() > 0 {
             //rather recover state
             self.recover_state();
+            for (genesis, chain) in genesis.iter().zip(self.chain.blocks()) {
+                if Hash::digest(genesis) != Hash::digest(chain) {
+                    warn!("Found a saved chain that is not compatible to our genesis.");
```
Comment (rtsisyk) [inline node/src/lib.rs]: Must not continue in this case. Please print both genesis blocks like in main().

Comment (vldm) [reply node/src/lib.rs]: @rtsisyk but then if some user has lost his genesis, then it being inpossible to load data. Are you sure that we need this behaviour?

Comment (rtsisyk) [reply node/src/lib.rs]: Who do you mean by "lost his genesis"?

Comment (vldm) [reply node/src/lib.rs]: @rtsisyk For example use binary with wrong genesis.

File: node/src/lib.rs [hunk 2]
```diff
@@ -361,6 +361,19 @@ impl NodeService {
         if self.chain.blocks().len() > 0 {
             //rather recover state
             self.recover_state();
+            for ((height, genesis), chain) in genesis.iter().enumerate().zip(self.chain.blocks()) {
+                let genesis_hash = Hash::digest(genesis);
+                let chain_hash = Hash::digest(chain);
+                if genesis_hash != chain_hash {
+                    warn!(
```
Comment (rtsisyk) [inline node/src/lib.rs]: error, not warn!

**Pull request merged**